content/en/blogs/argo-cd-a-tool-for-devops.md

```diff
@@ -0,0 +1,210 @@
+---
+
+In this post, I'll show you how Argo CD betters Kubernetes DevOps process. Before we begin practicing, let's look at some background information first.
+
+## Argo CD Capability
+
+### GitOps: The origin
+
+As you may already know, Weaveworks published a post titled [GitOps - Operations by Pull Request](https://www.weave.works/blog/gitops-operations-by-pull-request) in 2017. Alexis, the author, introduced a way of deployment by using Git as source of truth.
+
+In GitOps practices, we need to define and manage software infrastructures in Git repositories. Software infrastructures involve not only infrastructures like IaaS and Kubernetes but also applications. Everyone can make modifications to software infrastructures by submitting a Pull Request, and an automated program will perform those modifications.
+
+Thus, everyone will be able to focus on developing new features instead of worrying about operation and maintenance including installation, modification, and migration.
+
+### Argo CD: GitOps coming true
+
+Argo CD is a GitOps continuous delivery (CD) tool on top of Kubernetes. Here is the schematic diagram from Argo CD community.
+
+![argo-schematics](/images/blogs/en/argo-cd-a-tool-for-devops/argo-schematics.png)
+
+You can see from the diagram that:
+
+1. Argo CD pulls application configurations from Git Repo and deploys applications in Kubernetes clusters.
+2. If someone wants to add a new feature, he or she just needs to submit a Pull Request to Git Repo to modify application deployment configurations. Then, wait for merging.
+3. Once the Pull Request gets merged, Argo CD will run updating per Webhook triggers.
+4. Applications updated and notifications sent.
+
+To sum it up, automated operation and maintenance and continuous delivery. Simple and easy to understand, right?
+
+### Argo CD: Powerful and extensible
+
+The above functions will be able to cover ordinary Kubernetes operation and maintenance. What if things get more complicated? We cannot escape from scenarios involving multi-cloud, multi-platform, and multi-middleware.
+
+![argocd-2](/images/blogs/en/argo-cd-a-tool-for-devops/argocd-2.png)
+
+Argo CD defines four components in its processing logic:
+
+- Event Source: integrate various event messages.
+- Sensor: switch messages to triggering actions.
+- Eventbus: message subscription routing system.
+- Trigger: trigger external actions.
+
+As for operation and maintenance folks, you just need to focus on two aspects:
+
+- Events Argo CD can process: AMQP, AWS SNS, AWS SQS, Cron Schedules, GCP PubSub, GitHub, GitLab, HDFS, File Based Events, Kafka, Minio, NATS, MQTT, K8s Resources, Slack, NetApp StorageGrid, Webhooks, Stripe, NSQ, Emitter, Redis, Azure Events Hub.
+- Actions Argo CD can perform: Argo Workflows, Standard K8s Objects, HTTP Requests, AWS Lambda, NATS Messages, Kafka Messages, Slack Notifications, Argo Rollouts CR, Custom/Build Your Own Triggers, Apache OpenWhisk.
```

Comment:
Custom/Build Your Own Triggers, Apache OpenWhisk.
->
Custom/Build Your Own Triggers, and Apache OpenWhisk.